### PR TITLE
adding oscars recipe

### PR DESCRIPTION
**Pull request by kbeeperez**
implements: https://github.com/nsls-ii-forge/oscars-feedstock

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by kbeeperez** [comment]
Attn @mrakitin

**Review by mrakitin — APPROVED** [review]
Looks good to me, thanks @kbeeperez!

@conda-forge/staged-recipes, I agree to be a maintainer of this feedstock/recipe.

@dhidas, do you wish to co-maintain the conda-forge package for your project?

File: recipes/oscars/meta.yaml
Comment (dopplershift) [inline recipes/oscars/meta.yaml:20]: Pretty sure the recipe needs a compiler on Windows as well...

File: recipes/oscars/meta.yaml
Comment (dopplershift) [inline recipes/oscars/meta.yaml:33]: Might want to consider some more import tests. `setup.py` lists Windows build files, but you're not asking for a compiler on Windows--I'd expect the package to be broken.

File: recipes/oscars/meta.yaml
Comment (dopplershift) [inline recipes/oscars/meta.yaml:26]:
```suggestion
    - matplotlib-base
```

**Comment by dopplershift** [comment]
Does this package need CUDA support?

File: recipes/oscars/meta.yaml
Comment (mrakitin) [reply recipes/oscars/meta.yaml:20]: Yes, I agree. I don't remember exactly why I excluded it in https://github.com/nsls-ii-forge/oscars-feedstock.
@kbeeperez, let's get the compiler for all systems unconditionally.

I think the selector can be removed next to the `mpi4py` dependency too.

**Comment by mrakitin** [comment]
> Does this package need CUDA support?

It might, but @dhidas is in a better position to comment on that.

File: recipes/oscars/meta.yaml
Comment (mrakitin) [inline recipes/oscars/meta.yaml:20]:
```suggestion
    - {{ compiler('cxx') }}
```

File: recipes/oscars/meta.yaml
Comment (mrakitin) [inline recipes/oscars/meta.yaml:27]:
```suggestion
    - mpi4py
```

File: recipes/oscars/meta.yaml
Comment (mrakitin) [reply recipes/oscars/meta.yaml:33]: I like the idea. I think all entries from https://github.com/dhidas/OSCARS/blob/1f0823c3a8c9637b4c28e0afffefd361a62d8136/oscars/__init__.py#L1 can be added to the import test block.

**Review by mrakitin — COMMENTED** [review]
Left a couple of suggestions.

**Comment by kbeeperez** [comment]
@conda-forge/staged-recipes should be good to go now.

**Comment by mrakitin** [comment]
@dopplershift, it seems to be ready to go, as all additional tests passed for all 3 platforms.

> Does this package need CUDA support?

For that, what would we need to add to enable CUDA support? I guess it can be handled on the feedstock side once it's merged.

**Comment by dopplershift** [comment]
@mrakitin CUDA isn't really my wheelhouse, so I'm not really sure. I just know it usually takes a bit more effort to get everything building and linking right. You can certainly always fix things up in the feedstock if you find something isn't working how you want with the conda-forge packages.

File: recipes/oscars/meta.yaml
Comment (dopplershift) [inline recipes/oscars/meta.yaml:43]: This actually looks more like the [EPICS](https://epics.anl.gov/license/open.php) license (though with a minor extra clause).
```suggestion
  license: EPICS
  license_family: BSD
```

Comment (mrakitin) [reply recipes/oscars/meta.yaml]: Sounds reasonable, thanks!

**Review by dopplershift — APPROVED** [review]

**Comment by mrakitin** [comment]
Thanks, @dopplershift!

**Pull request merged**
